File: src/my_project/easy_problems/from101to150/find_nearest_point_same_x_y.py
from typing import List 

class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:

        answer: List[int] = list()
        answer = points.copy()
        flag: bool = False
        for item in points:
            if item[0] == x or item[1] == y:
                flag = True 
        
        if flag == False:
            return -1
        
    
        def distance_manhattan(point: List[int]):
            return abs(point[0] - x) + abs(point[1] - y)

        points.sort(key= lambda point: distance_manhattan(point))
        print(points)
        return answer.index(points[0])

solution = Solution()

print(solution.nearestValidPoint(x = 3, y = 4, points = [[1,2],[3,1],[2,4],[2,3],[4,4]]))

print(solution.nearestValidPoint(x = 3, y = 4, points = [[2,3]]))

print(solution.nearestValidPoint(5,1 ,[[1,1],[6,2],[1,5],[3,1]]))

File: my_project_typescript/easy_problems_typescript/from1to50/findNearestPointSameXY.ts
function nearestValidPoint(x: number, y: number, points: number[][]): number {

    let answer: Array<any> = []
    let flag: boolean = false 
    answer = Array.from(points)
    answer.reverse()

    for (let item of points){
        if (item[0]===x || item[1]===y){
            flag = true
        }
    }

    if (flag ===false){
        return -1
    }

    function distance_manhattan(point: Array<any>){
        return Math.abs(point[0]-x) + Math.abs(point[1]-y)
    }

    answer = answer.sort((a,b) => distance_manhattan(a) <= distance_manhattan(b) ?-1:1)

    console.log(answer)

    for (let i = 0; i < points.length; i++){
        if (points[i][0] === answer[0][0] && points[i][1] === answer[0][1]){
            return i
        }
        
    }
    return 0
};


console.log(nearestValidPoint(3, 4,  [[1,2],[3,1],[2,4],[2,3],[4,4]]))